src/Core/src/App.cs

```diff
@@ -14,6 +12,7 @@ protected App()
 		{
 			if (Current != null)
```

Comment:
I know this is different ideas, but I think that we shouldn't have a static `Current` on the Application object - this makes it hard to test nicely - but rather the current should be on the Android `Application` or the iOS `UIAppDelegate` part. This way, we can have multiple apps in unit tests, and we never lose anything since we just moved the static property. Having a static on something that is going to be part of a resolved service is a bit weird.